x-pack/elastic-agent/docs/elastic-agent-configuration.asciidoc

```diff
@@ -0,0 +1,113 @@
+[[elastic-agent-configuration]]
+= Configuration settings
+
+experimental[]
+
+By default {agent} runs in standalone mode to ingest system data and send it to
+a local {es} instance running on port 9200. It uses the demo credentials of the
+`elastic` user. It's also configured to monitor all {beats} managed by the agent
+and send the {beats} logs and metrics to the same {es) instance.
+
+To alter this behavior, configure the output and other configuration settings:
+
+* <<elastic-agent-output-configuration>>
+* <<elastic-agent-monitoring-configuration>>
+* <<elastic-agent-datasource-configuration>>
+
+[float]
+[[elastic-agent-output-configuration]]
+== Output settings
+
+Specify one or more outputs. Specifying multiple outputs allows you to pair
+each data source with a different output. 
+
+IMPORTANT: {agent} currently works with the {es} output only.
+
+Example output configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------------
+outputs:
+  default:
+    type: elasticsearch
+    hosts: [127.0.0.1:9200]
+    username: elastic
+    password: changeme
+
+  monitoring:
+    type: elasticsearch
+    api_key: VuaCfGcBCdbkQm-e5aOx:ui2lp2axTNmsyakw9tvNnw
+    hosts: ["localhost:9200"]
+    ca_sha256: "7lHLiyp4J8m9kw38SJ7SURJP4bXRZv/BNxyyXkCcE/M="
+-------------------------------------------------------------------------------------
+
+This example configures two outputs: `default` and  `monitoring`.
+Notice that they use different authentication methods. The first one uses a
+username and password pair, and the second one contains an API key.
+
+[NOTE]
+==============
+A default output configuration is required.
+==============
+
+[float]
+[[elastic-agent-monitoring-configuration]]
+== {beats} monitoring settings
+
+{agent} monitors {beats} by default. To disable or change monitoring 
+settings, set options under `settings.monitoring`:
+
+[source,yaml]
+-------------------------------------------------------------------------------------
+settings.monitoring:
+  # enabled turns on monitoring of running processes
+  enabled: true
+  # enables log monitoring
+  logs: true
+  # enables metrics monitoring
+  metrics: true
+  # specifies output to be used
+  use_output: monitoring
+-------------------------------------------------------------------------------------
+
+
+To disable monitoring, set `settings.monitoring.enabled` to `false`. When set to
+`false`, {beats} monitoring is turned off, and all other options in this section
+are ignored.
+
+To enable monitoring, set `settings.monitoring.enabled` to `true`. Also set the
+`logs` and `metrics` settings to control whether logs, metrics, or both are
+collected. If neither setting is specified, monitoring is disabled. Set
+`use_output` to specify the output to which monitoring events are sent.
+
+[[elastic-agent-datasource-configuration]]
+== Datasource settings
+
+//TODO: provide more context here about what a data source is and how it's used.
```

Comment:
Datasource will be removed in 7.9, so we can be high level here.

Reply:
Good. I found the concept to be a bit nebulous. Fewer concepts = easier to understand. yay!

Reply:
You were not alone :) cc @ruflin